File: Step by Step/deep_model.py
    def deep_model(X,parameters):
        """
        Takes X and parameters, respectively the data and the initialized parameters from init_param() function.
        Returns AV and caches, the activation value at the end of the architecture, and the cached values of every layer.
        """
        caches=[]
        A=X
        L=len(parameters)//2 #The floor division gives the number of layers in the network
        for l in range(1,L):
            A_prev=A
            A,cache=for_activation(A_prev,parameters['W'+str(l)],parameters['b'+str(l)],activation='relu')
            caches.append(cache)
        AV,cache=for_activation(A,parameters['W'+str(l)],parameters['b'+str(l)],activation='sigmoid')
        caches.append(cache)
        return AV,caches

File: deep_model.R
source("for_prop.R")
deep_model<-function(X,parameters){
  caches<-list()
  A<-X
  L<-length(parameters)%/%2
  for (l in 1:(L)){
    if (l==L){
      A_prev<-A
      calc<-for_activation(A_prev,parameters[[paste0("W",L)]],parameters[[paste0("b",L)]],activ='sigmoid')
      AV<-calc[[1]]
      cache<-calc[[2]]
      caches[l]<-list(cache)
    } else {
    A_prev<-A
    calc<-for_activation(A_prev,parameters[[paste0("W",l)]],parameters[[paste0("b",l)]],activ='relu')
    A<-calc[[1]]
    cache<-calc[[2]]
    caches[l]<-list(cache)
    }
  }
  return(list(AV,caches))
}
